### PR TITLE
Refactor governance events

### PR DESCRIPTION
**Pull request by Fraccaman**
## Describe your changes
Refactor governance events.

## Indicate on which release or other PRs this topic is based on
Based on #3102 

## Checklist before merging to `draft`
- [x] I have added a changelog
- [x] Git history is in acceptable state

### CODE REVIEW

**Review by sug0 — COMMENTED** [review]
just left a comment

File: crates/governance/src/event.rs [hunk 1]
```diff
@@ -63,68 +54,55 @@ pub struct ProposalEvent {
...
-    PgfSteward {
-        /// The outcome of the proposal.
-        result: bool,
-    },
-    /// PGF payments governance proposal.
-    PgfPayments {
-        /// The outcome of the proposal.
-        result: bool,
+        has_proposal_code: bool,
+        is_proposal_code_successful: bool,
```
[the remark this answers is not in the document]
Comment (Fraccaman) [reply crates/governance/src/event.rs:102]: agree, but I think its good enough for now

**Review by sug0 — DISMISSED** [review]
ok, that was my only real complaint.

btw make sure to rebase again, when this eventually is ready. I've changed the base a few times due to review comments, already, so your base has diverged from mine.

**Comment by Fraccaman** [comment]
@sug0 can we cherry pick my commit to ur branch? its gonna make it easier

**Pull request merged**